api/src/main/java/com/epam/pipeline/manager/notification/NotificationAspect.java

```diff
@@ -89,5 +96,13 @@ public void notifyRunStatusChanged(JoinPoint joinPoint, PipelineRun run) {
                      run.getPodId(), run.getPipelineName(), run.getVersion(), run.getStatus());
         notificationManager.notifyRunStatusChanged(run);
     }
+
+    private String getStatusReason(final PipelineRun run) {
```

Comment:
@SilinPavel Maybe we could just pass the status in `PipelineRun.stateReasonMessage`? It is also not the best solution.